src/mercedestrenz/data.py

```diff
@@ -52,12 +55,73 @@ def listing_search(data, budget=[0, np.Inf], model = "any", sort_feature = "odom
     
     Examples
     --------
-    >>> from mercedestrenz.mercedestrenz import listing_search
-    >>> result = listing_search(data, budget=[2000, 20000], model = "any", sort_feature = "odometer", ascending = True)
-    >>> result.head()
-    #to be completed with exmaple output
+    >>> # search listings within a price range
+    >>> listing_search(data, budget=[2000, 20000], model = "any", sort_feature = "odometer_mi", ascending = True)
+    >>> # search listings for a specific model and below a maximum price
+    >>> listing_search(data, budget=20000, model = "glk", sort_feature = "odometer_mi", ascending = True)
     """
-    pass
 
+    # ======= Unit tests ==========
+    # check input data is a pd.dataframe
+    if type(data) != pd.DataFrame:
+        raise Exception("The input dataset is not of Pandas DataFrame format")
+
+    # check budget is either a flort/int or a list of float/int
+    if type(budget) not in [float, int]: 
+        if len(budget) == 2:
+            for elem in budget:
+                if type(elem) not in [float, int]:
+                    raise Exception("The budget range boundaries should be floats or integers")
+        else:
+            raise Exception("Please specify your maximum budget using a number, or a range of budge using a list of numeric values.")
+
+    # model= parameter input should be a string
+    if type(model) != str:
+            raise Exception("The model parameter should have a string as the input")
+    
+    # The string should correspond to a specific car model in the model column
+    if ((data['model'].eq(model)).any() or model == 'any') == False:
```

Comment:
- optional variation that might improve readability. 
```suggestion
    if ((model in data['model'].unique()) or model == 'any') == False:
```